components/net/http_loader.rs

```diff
@@ -515,7 +516,9 @@ pub fn modify_request_headers(headers: &mut Headers,
 
     set_default_accept(headers);
     set_default_accept_encoding(headers);
-    set_request_cookies(doc_url.clone(), headers, cookie_jar);
+    if load_data.include_third_party_credentials {
+        set_request_cookies(doc_url.clone(), headers, cookie_jar);
```

Comment:
You can add this comment: `// https://fetch.spec.whatwg.org/#concept-http-network-or-cache-fetch step 11`